services/filemanager/filemanager.go

```diff
@@ -135,9 +123,11 @@ func GetProviderConfigFromEnv() map[string]interface{} {
 	return providerConfig
 }
 
-func getSafeTimeout(timeout *time.Duration) time.Duration {
-	if timeout == nil || *timeout == 0 {
-		return time.Second * 120
+func getBatchRouterDurationConfig(key, destType string, defaultValueInTimescaleUnits int64, timeScale time.Duration) time.Duration {
+	destOverrideFound := config.IsSet("BatchRouter." + destType + "." + key)
+	if destOverrideFound {
+		return config.GetDuration("BatchRouter."+destType+"."+key, defaultValueInTimescaleUnits, timeScale)
+	} else {
+		return config.GetDuration("BatchRouter."+key, defaultValueInTimescaleUnits, timeScale)
```

Comment:
This function accepts a lot of arguments that don't actually change. We could use this instead
```suggestion
func getTimeout(destType string) time.Duration {
	key := fmt.Sprintf("BatchRouter.%s.timeout", destType)
	destOverrideFound := config.IsSet(key)
	if destOverrideFound {
		return config.GetDuration(key, 120, time.Second)
	} else {
		return config.GetDuration("BatchRouter.timeout", 120, time.Second)
	}
}
```

Reply:
The only problem with this is different managers can't have different defaults.

Reply:
wasn't aware we had such a requirement, but we can still include the default `time.Duration` in the function's arguments :)